### PR TITLE
Feature/jess th username case

### LINKED ISSUE #2918: Excluding Usernames for TDTM Triggers does not work if the Usernames case does not exactly match
The issue is that Set.contains() is case-sensitive. Can be fixed easily by forcing everything to upper or lower case before using contains().
https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/TDTM_ObjectDataGateway.cls#L74

### PR DESCRIPTION
**Pull request by jesslopez-sf**
# Critical Changes

# Changes
Prior to this fix, usernames entered in a Trigger Handler record's Usernames to Exclude field needed to be case-sensitive.
# Issues Closed
Fixes #2918

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by jesslopez-sf** [comment]
**lurch: attach W-022118

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000009KBq2AAG" target="_blank">W-022118</a>

File: src/classes/TDTM_Config_API_Test.cls [hunk 1]
```diff
@@ -42,17 +42,17 @@ private class TDTM_Config_API_Test {
...
+    static void triggersAreDisabledForSpecificContext() {
         //skip the test if Advancement is installed
-        if(ADV_PackageInfo_SVC.useAdv()) return;
+        if (ADV_PackageInfo_SVC.useAdv()) return;
         
         TDTM_DefaultConfig.startTestDataInit();
         Test.startTest();
 
-        Contact con1 = new Contact(LastName='foo');
+        Contact con1 = new Contact(LastName = 'foo');
```
Comment (force2b) [inline src/classes/TDTM_Config_API_Test.cls]: Minor, but would you mind changing all the places in the test class that create a new Contact to use `UTIL_UnitTestData_TEST.getContact();`?

For example: `Contact con1 = UTIL_UnitTestData_TEST.getContact();`

Comment (jesslopez-sf) [reply src/classes/TDTM_Config_API_Test.cls]: All set. Thank you, @force2b ! I rarely look at this side of the codebase, so I just learned something new. :)

Comment (jesslopez-sf) [reply src/classes/TDTM_Config_API_Test.cls]: All set. Thanks, @force2b ! I rarely look at this side of the codebase, so I learned something new. :)

File: src/classes/TDTM_ObjectDataGateway.cls [hunk 2]
```diff
@@ -46,33 +46,33 @@ public without sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGa
...
-        if(rowCount == 0) {
+        if (rowCount == 0) {
             return true;
         }
         return false;
     }
 
     /*******************************************************************************************************
      * @description Returns the classes that will be called for the specified ojbect and action. Returned list 
-     *              is sorted by load order. Filters out in-active handlers and any where the list of
```
Comment (Julian88Tex) [inline src/classes/TDTM_ObjectDataGateway.cls]: @jesslopez-sf love the spelling corrections :)

**Comment by Julian88Tex** [comment]
@force2b @jesslopez-sf I've completed testing with all tests passing.

I tested using the RD_RecurringDonations_TDTM & OPP_OpportunityContactRoles_TDTM with a variety of scenarios and with bulk data creation.

It appears I'm unable to merge this PR due to not having authorization (not sure why) but feel free to go ahead and merge it.

**Comment by force2b** [comment]
Thanks @Julian88Tex. The merge freeze is still active until 4pm pst, but you're free to merge after it's lifted.

**Comment by Julian88Tex** [comment]
@force2b Ah thanks for the reminder for some reason I thought it was mid-day but good to know thanks!

**Comment by Julian88Tex** [comment]
@force2b any post or notification I should look out for, for end of merge freeze or am I good assuming I can click the button?

**Comment by Julian88Tex** [comment]
@jesslopez-sf @force2b this change has been merged :-)

**Pull request merged**